src/transformers/generation/logits_process.py

```diff
@@ -1572,14 +1579,15 @@ def __init__(
         self.eos_token_id = eos_token_id
 
     @add_start_docstrings(LOGITS_PROCESSOR_INPUTS_DOCSTRING)
-    def __call__(self, input_ids: torch.LongTensor, scores: torch.FloatTensor) -> torch.FloatTensor:
-        cur_len = input_ids.shape[-1]
+    def __call__(self, input_ids: torch.LongTensor, scores: torch.FloatTensor, cur_len: int) -> torch.FloatTensor:
```

Comment:
Looking at this one, there is a chance we need to make `cur_len` a tensor, and not an `int`. If torch's compilation behaves anything like TF's, then integers will trigger a new compilation each time it is called with a different value 🤔 

Try to confirm that a recompilation is not happening as you call this one with different `cur_len`. If a recompilation is happening, try changing `cur_len` to a tensor. Avoiding recompilations is crucial for performance.

Reply:
Yes, you are right, `cur_len` as int causes recompilation every time the processor is called. But when we pass it in as a tensor, it will raise en error when trying to get tensor's value and compare the current length.

There is actually a way to use `functorch.cond` and then we can avoid graph breaks and recompilation, but it does not work with inductor backend right now. Pytorch devs said that it will be ready to use and efficient around the end of Q1.